### PR TITLE
Fix orca’s flying sequences.

### LINKED ISSUE #7889: TD orca uses wrong sequences for movement
Currently it always uses frames 1-32, and only uses 33-64 if damaged.

The correct behavior would be to use frames 1-32 for idle (standing still in the air, landed) and frames 33-64 when moving.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10598289-td-orca-uses-wrong-sequences-for-movement?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Fixes #7889.  Conflicts with #7887 (sigh).

### CODE REVIEW

**Comment by Mailaender** [comment]
:+1: Works as promised. :white_check_mark: Adds a nice little detail.

**Comment by pchote** [comment]
Rebased on top of merged dependency.

File: OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+		public override void Tick(Actor self)
+		{
+			base.Tick(self);
+
+			var oldCachedPosition = cachedPosition;
+			cachedPosition = self.CenterPosition;
+
+			// Flying units set IsMoving whenever they are airborne, which isn't enough for our purposes
+			var isMoving = movement.IsMoving && !self.IsDead && (oldCachedPosition - cachedPosition).HorizontalLengthSquared != 0;
```
Comment (obrakmann) [inline OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs]: Can't you just compare `oldCachedPosition` and `cachedPosition`?

File: OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			if (isMoving ^ (DefaultAnimation.CurrentSequence.Name != info.MoveSequence))
+				return;
+
+			DefaultAnimation.ReplaceAnim(isMoving ? info.MoveSequence : "idle");
```
Comment (obrakmann) [inline OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs]: Can you un-hardcode that, too, please?

**Comment by reaperrr** [comment]
I tested this some days ago and it works fine, :+1: once the things obrakmann pointed out have been adressed.

File: OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs (hunk 1 above)
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs]: Nope - we want this to be false if it is taking off or landing.

File: OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs (hunk 2 above)
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/RenderUnitFlying.cs]: Fixed.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/395219f7935dd11ea0d81647c752c4475f801688%5E...395219f7935dd11ea0d81647c752c4475f801688)

**Pull request merged**